core/internal/httpserver/prometheus.go

```diff
@@ -0,0 +1,182 @@
+package httpserver
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/linkedin/Burrow/core/protocol"
+
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	consumerTotalLagGauge = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "burrow_kafka_consumer_lag_total",
+			Help: "The sum of all partition current lag values for the group",
+		},
+		[]string{"cluster", "consumer_group"},
+	)
+
+	consumerStatusGauge = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "burrow_kafka_consumer_status",
+			Help: "The status of the consumer group. It is calculated from the highest status for the individual partitions. Statuses are an index list from NOTFOUND, OK, WARN, ERR, STOP, STALL, REWIND",
+		},
+		[]string{"cluster", "consumer_group"},
+	)
+
+	consumerPartitionCurrentOffset = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "burrow_kafka_consumer_current_offset",
+			Help: "Latest offset that Burrow is storing for this partition",
+		},
+		[]string{"cluster", "consumer_group", "topic", "partition"},
+	)
+
+	consumerPartitionLagGauge = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "burrow_kafka_consumer_partition_lag",
+			Help: "Number of messages the consumer group is behind by for a partition as reported by Burrow",
+		},
+		[]string{"cluster", "consumer_group", "topic", "partition"},
+	)
+
+	topicPartitionOffsetGauge = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "burrow_kafka_topic_partition_offset",
+			Help: "Latest offset the topic that Burrow is storing for this partition",
+		},
+		[]string{"cluster", "topic", "partition"},
+	)
+)
+
+func (hc *Coordinator) handlePrometheusMetrics() http.HandlerFunc {
+	promHandler := promhttp.Handler()
+
+	return http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
+		for _, cluster := range listClusters(hc.App) {
+			for _, consumer := range listConsumers(hc.App, cluster) {
+				consumerStatus := getFullConsumerStatus(hc.App, cluster, consumer)
+
+				if consumerStatus == nil ||
+					consumerStatus.Status == protocol.StatusNotFound ||
+					consumerStatus.Complete < 1.0 {
```

Comment:
I am wondering what is the reason for this particular rule.
We have some partitions that have the completion percentage `0.1`. The cause of this seems to be related to the fact that we have some partitions that are empty and eventually - their offset expires(although I'm not sure - it might be something else on our side). The consequence is that the completion percentage for the consumer is less than 1 and it is not included in prometheus.